pkg/templates/nstemplatetiers/nstemplatetier_generator_whitebox_test.go

```diff
@@ -160,6 +160,10 @@ func TestNewNSTemplateTier(t *testing.T) {
 							_, _, err = decoder.Decode(asset, nil, tmplObj)
 							require.NoError(t, err)
 							assert.Equal(t, *tmplObj, ns.Template)
+
+							// Assert expected objects in the template
+							// Each template should have one Namespace and one RoleBinding object
+							require.Len(t, ns.Template.Objects, 2)
```

Comment:
if we know that each template should have at least one `Namespace` and one `RoleBinding` object, why we cannot verify it directly, instead of just checking the length...?

Reply:
Checking the objects inside the template is not that easy. We would need to process the template. Basically would need to bring the template processor from member operator repo here. So we would need to move it common and reuse everywhere. It’s not a trivial change.
A simple decoding doesn’t seem to be enough as well. Maybe I missed something though since I didn’t spent much time on it.
But we do proper testing of the roles and namespaces in our e2e tests anyway. So I think it’s not worth significantly complicate our unit tests for the cases which already covered by e2e tests.

Reply:
I see, but we could do a bit simpler check - either this:
```go
assert.Contains(t, string(ns.Template.Objects[1].Raw), `"kind":"RoleBinding","metadata":{"labels":{"provider":"codeready-toolchain"},"name":"user-edit"`)
```
or more safe:
```go
rbFound := false
for _, object := range ns.Template.Objects {
	if strings.Contains(string(object.Raw), `"kind":"RoleBinding","metadata":{"labels":{"provider":"codeready-toolchain"},"name":"user-edit"`) {
		rbFound = true
		break
	}
}
assert.True(t,rbFound, "the user-edit RoleBinding wasn't found in the namespace of the type", "ns-type", nsType)
```
take it just as an suggestion - it's not a must ;-)

Reply:
Done in https://github.com/codeready-toolchain/host-operator/pull/106/commits/8202e03f2c85262e9359ea749643eda44c4fdde0